bpm/src/main/java/org/jboss/pnc/bpm/causeway/ProductMilestoneReleaseManager.java

```diff
@@ -187,6 +199,8 @@ private void storeResult(ProductMilestone milestone, MilestoneReleaseResultRest
             milestoneRepository.save(milestone);
 
             removeCurrentFlagFromMilestone(milestone);
+
+            productMilestoneCloseResultEvent.fire(mapper.toDTO(productMilestoneRelease));
```

Comment:
What about when milestone closing fails? We definitely need to send a notification in that case.

Reply:
My point still stands. You are firing an event only when `result.getReleaseStatus().getMilestoneReleaseStatus() == MilestoneCloseStatus.SUCCEEDED`, which means, only when the closing succeeded. You don't take care of other cases, (f.e. exception thrown, FAILED result...) and users need to be notified.